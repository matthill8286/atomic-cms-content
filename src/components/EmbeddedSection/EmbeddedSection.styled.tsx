import {css, styled} from '@matthill8286/atomic-ui';
import {CustomSection} from '../CustomSection';

export const StyledPaddedEmbed = styled.div(
  ({theme}) =>
    css`
      padding: ${theme.spacing.base.md};
      position: absolute;
      top: 0;
      left: 0;
      bottom: 0;
      right: 0;
      width: 100%;
      height: 100%;
    `
);

export const StyledPaddedEmbedSection = styled(CustomSection)(
  ({theme}) =>
    css`
      padding: ${theme.spacing.base.md};
      position: relative;
      overflow: hidden;
      width: 100%;
      padding-top: 56.25%;
      margin-bottom: calc(${theme.spacing.base.xl}* 2);
    `
);
