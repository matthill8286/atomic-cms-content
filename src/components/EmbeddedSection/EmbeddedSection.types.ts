import {ThemeColors} from '@matthill8286/atomic-ui';
import {Raw, RichTextBlock} from '../../types/richtext/graphcms-richtext';

export interface EmbeddedSectionProps {
  embeddedSection: Raw;
  sectionColor?: ThemeColors;
  title?: string;
  sectionHeight?: string;
}
