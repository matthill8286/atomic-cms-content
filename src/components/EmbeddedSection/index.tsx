import React, { FC } from 'react'
import { Grid, Row, Cell, Heading } from '@matthill8286/atomic-ui'
import { ConfigurableGraphCmsHtmlSerializer, Serializer } from '../Serializer'
import { CustomSection } from '../CustomSection'
import { StyledPaddedEmbedSection, StyledPaddedEmbed } from './EmbeddedSection.styled'
import { EmbeddedSectionProps } from './EmbeddedSection.types'

export const EmbeddedSection: FC<EmbeddedSectionProps> = ({
  embeddedSection,
  sectionColor,
  title,
}) => {
  return (
    <CustomSection color={sectionColor}>
      <Grid>
        <Row noMargin>
          <Cell columns={12}>
            <Heading scale="level-2" textAlign="center" margin="xxxxl 0 sm 0">
              {title}
            </Heading>
          </Cell>
        </Row>
        <Row noMargin>
          <Cell columns={12}>
            <StyledPaddedEmbedSection overrideOver>
              <StyledPaddedEmbed>
                {embeddedSection &&
                  Serializer(
                    embeddedSection.raw,
                    // @ts-ignore
                    ConfigurableGraphCmsHtmlSerializer({
                      textAlign: 'center',
                    })
                  )}
              </StyledPaddedEmbed>
            </StyledPaddedEmbedSection>
          </Cell>
        </Row>
      </Grid>
    </CustomSection>
  )
}
