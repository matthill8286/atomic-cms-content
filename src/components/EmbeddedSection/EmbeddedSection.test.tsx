import React from 'react';
import {renderWithTheme} from '@matthill8286/atomic-ui';
import {EmbeddedSection} from '../EmbeddedSection';

xdescribe('EmbeddedSection', () => {
  it.skip('shows an embedded section', () => {
    const tree = renderWithTheme(
      <EmbeddedSection
        embeddedSection={{raw: {type: 'heading-one', children: []}}}
        sectionColor="primary"
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
