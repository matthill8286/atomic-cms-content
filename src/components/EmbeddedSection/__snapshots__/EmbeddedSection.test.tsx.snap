// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbeddedSection shows an embedded section 1`] = `
.c5 {
  margin: 64px 0 16px 0;
  font-size: 30px;
  line-height: 28px;
  font-family: Montserrat,Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: 0.4px;
  -moz-letter-spacing: 0.4px;
  -ms-letter-spacing: 0.4px;
  letter-spacing: 0.4px;
  text-align: center;
}

.c10 {
  margin: 32px 0;
  font-size: 34px;
  line-height: 40px;
  font-family: Montserrat,Arial,sans-serif;
  font-weight: 700;
  -webkit-letter-spacing: 0.4px;
  -moz-letter-spacing: 0.4px;
  -ms-letter-spacing: 0.4px;
  letter-spacing: 0.4px;
  text-align: center;
}

.c0 {
  width: 100%;
  left: 0;
  right: 0;
  background: #00C8DC no-repeat center center;
  background-size: cover;
}

.c6 {
  width: 100%;
  left: 0;
  right: 0;
  background: no-repeat center center;
  background-size: cover;
}

.c4 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  box-sizing: border-box;
  margin: calc(16px / 2);
  width: calc(100% - 16px);
  width: calc(100% - 16px);
}

.c2 {
  box-sizing: border-box;
  margin: 0 auto;
  max-width: 1472px;
  padding: 0 16px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  margin: 0;
  margin-bottom: 0;
}

.c1 {
  background-color: #00C8DC;
}

.c1 ul a,
.c1 ol a,
.c1 p a {
  border-bottom: 1px solid;
  border-color: #00C8DC;
  color: #00C8DC;
}

.c7 {
  overflow: unset !important;
}

.c7 ul a,
.c7 ol a,
.c7 p a {
  border-bottom: 1px solid;
  border-color: #00C8DC;
  color: #00C8DC;
}

.c9 {
  padding: 24px;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  width: 100%;
  height: 100%;
}

.c8 {
  padding: 24px;
  position: relative;
  overflow: hidden;
  width: 100%;
  padding-top: 56.25%;
  margin-bottom: calc(40px* 2);
}

@media (min-width:752px) {

}

@media (min-width:1008px) {

}

@media (min-width:752px) {

}

@media (min-width:1008px) {

}

@media (min-width:752px) {

}

@media (min-width:1008px) {

}

@media (min-width:752px) {

}

@media (min-width:1008px) {

}

@supports (display:grid) {
  .c4 {
    margin: 0;
    width: auto;
    grid-column-end: span 4;
  }
}

@supports (display:grid) {
  .c4 {
    width: auto;
    grid-column-end: span 4;
  }
}

@media (min-width:512px) {
  .c4 {
    display: block;
    margin: calc(16px / 2);
    width: calc(50% - 16px);
    width: calc(100% - 16px);
  }
}

@media (min-width:752px) {
  .c4 {
    display: block;
    margin: calc(24px / 2);
    width: calc(50% - 24px);
    width: calc(100% - 24px);
  }
}

@media (min-width:1008px) {
  .c4 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    width: calc(100% - 24px);
  }
}

@media (min-width:1232px) {
  .c4 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    width: calc(100% - 24px);
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {

}

@media (min-width:512px) {
  .c2 {
    padding: 0 24px;
  }
}

@media (min-width:752px) {
  .c2 {
    padding: 0 32px;
  }
}

@media (min-width:1008px) {
  .c2 {
    padding: 0 48px;
  }
}

@media (min-width:1232px) {
  .c2 {
    padding: 0 64px;
  }
}

@supports (display:grid) {
  .c3 {
    display: grid;
    grid-gap: 16px;
    grid-template-columns: repeat(4,minmax(0,1fr));
  }
}

@media (min-width:512px) {

}

@media (min-width:752px) {

}

@media (min-width:1008px) {

}

@media (min-width:1232px) {

}

<section
  className="c0 c1"
  color="primary"
>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <h2
          className="c5"
          fontFamily="default"
          fontSize="xxl"
          spacing="base"
        />
      </div>
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <section
          className="c6 c7 c8"
          color="transparent"
        >
          <div
            className="c9"
          >
            <h1
              className="c10"
              fontFamily="default"
              fontSize="xxxl"
              spacing="base"
            />
          </div>
        </section>
      </div>
    </div>
  </div>
</section>
`;
